[docs]: update building modules section to reflect ADR31 (#7702)
cosmos/cosmos-sdk: docs/building-modules/intro.md

On top of this core, the Cosmos SDK enables developers to build modules that implement the business logic of their application. In other words, SDK modules implement the bulk of the logic of applications, while the core does the wiring and enables modules to be composed together. The end goal is to build a robust ecosystem of open-source SDK modules, making it increasingly easier to build complex blockchain applications. 

> SDK Modules can be seen as little state-machines within the state-machine. They generally define a subset of the state using one ore multiple `KVStore` in the [main multistore](../core/store.md), as well as a subset of [`message` types](./messages-and-queries.md#messages). These `message`s are routed by one of the main component of SDK core, [`baseapp`](../core/baseapp.md), to the [`Msg` service](./msg-services.md) of the module that define them. 

Reviewer: ```suggestion
SDK Modules can be seen as little state-machines within the state-machine. They generally define a subset of the state using one or more `KVStore`s in the [main multistore](../core/store.md), as well as a subset of [message types](./messages-and-queries.md#messages). These messages are routed by one of the main component of SDK core, [`baseapp`](../core/baseapp.md), to the [`Msg` service](./msg-services.md) of the module that define them. 
```

> SDK Modules can be seen as little state-machines within the state-machine. They generally define a subset of the state using one ore multiple `KVStore` in the [main multistore](../core/store.md), as well as a subset of [`message` types](./messages-and-queries.md#messages). These `message`s are routed by one of the main component of SDK core, [`baseapp`](../core/baseapp.md), to the [`handler`](./msg-services.md#handler-type) of the module that define them. 

Reviewer 2: ```suggestion
SDK Modules can be seen as little state-machines within the state-machine. They generally define a subset of the state using one ore multiple `KVStore` in the [main multistore](../core/store.md), as well as a subset of [`message` types](./messages-and-queries.md#messages). These `message`s are routed by one of the main component of SDK core, [`baseapp`](../core/baseapp.md), to the [`Msg` service](./msg-services.md) of the module that define them. 
```

Modules are by convention defined in the `.x/` subfolder (e.g. the `bank` module will be defined in the `./x/bank` folder). They generally share the same core components:

- A [`Msg` service](./messages-and-queries.md#messages) to trigger state-transitions. 

> - A [`handler`](./msg-services.md#handler-type) used to process messages when they are routed to the module by [`baseapp`](../core/baseapp.md#message-routing). 

Reviewer 2: Maybe we should remove `handler` here? It's in the intro of this section.
